test/prune-test.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('./util/assert');
+const consensus = require('../lib/protocol/consensus');
+const FullNode = require('../lib/node/fullnode');
+const Network = require('../lib/protocol/network');
+const network = Network.get('regtest');
+
+const node = new FullNode({
+  network: 'regtest',
+  apiKey: 'foo',
+  walletAuth: true,
+  memory: true,
+  workers: true,
+  plugins: [require('../lib/wallet/plugin')],
+  prune: true
+});
+
+const {NodeClient, WalletClient} = require('bclient');
+
+const nclient = new NodeClient({
+  port: network.rpcPort,
+  apiKey: 'foo'
+});
+
+const wclient = new WalletClient({
+  port: network.walletPort,
+  apiKey: 'foo'
+});
+
+let wallet = null;
+
+describe('Pruned node', function() {
+  this.timeout(60000);
+
+  it('should open node', async () => {
+    consensus.COINBASE_MATURITY = 0;
+    await node.open();
+    await nclient.open();
+    await wclient.open();
+  });
+
+  it('should create wallet', async () => {
+    const info = await wclient.createWallet('test');
+    assert.strictEqual(info.id, 'test');
+    wallet = wclient.wallet('test', info.token);
+    await wallet.open();
+  });
+
+  it('should generate 1000 blocks', async () => {
+    const addr = await wallet.createAddress('default');
+    assert(await nclient.execute('generatetoaddress', [100, addr.address]));
```

Comment:
Why not just 1000 at once?

Reply:
Well - I was getting timeout errors. Not from the testing suite (because timeout is st to `60000` on line 37) but from actual bclient. And not even on local, just on Travis. SO it's a bit of a hacky work-around. I'm using bclient to generate blocks in this test, whereas other blockchain tests in `bcoin` use custom functions like `mineBlock` and `CPUminer`. I was lazy in writing this test with bclient but also figure, we're getting a tiny bit of extra coverage this way because there aren't any other tests for this API call.